### ISSUE TITLE
rancher cli inconsistency between using --env env name or id, with users other than admin

### ISSUE THREAD

**Issue by aemneina** [rancher/rancher#9988]
**Rancher versions:**
rancher/server: 1.6.10
rancher/cli: 0.6.4

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
cattle

**Steps to Reproduce:**
1. create a user
2. add that user to various environments
3. query stacks with rancher --env <env-name> stack ls

**Results:**
when using env id this works, but env name throws something like: `FATA[0000] Not found: Default`

**Comment by yasker** [comment]
Reminder for QA:

Test auth off, auth on, admin user, non-admin user.  test `rancher env ls` and `rancher env ls -a`

**Comment by deniseschannon** [comment]
Available with CLI v0.6.5-rc5.

**Comment by kingsd041** [comment]
Tested with rancher-server version- v1.6.10 with rancher/cli:v0.6.5-rc5
Test passed.

```
#  rancher help | grep Version
Version: v0.6.5-rc5

#  rancher --env hailong stack ls
ID        NAME           STATE     CATALOG   SERVICES   SYSTEM    DETAIL
1st15     hailongstack   healthy             0          false
#  rancher --env default stack ls

ID        NAME       STATE     CATALOG   SERVICES   SYSTEM    DETAIL
1st10     default2   healthy             0          false

#  rancher env ls
ID        NAME      ORCHESTRATION   STATE     CREATED
1a5       Default   cattle          active    2017-11-07T02:17:03Z
1a23      hailong   cattle          active    2017-11-07T03:39:01Z
```
